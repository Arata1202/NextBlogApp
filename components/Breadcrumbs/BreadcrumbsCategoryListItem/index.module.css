.tag {
  border-radius: var(--border-radius);
  font-size: 14px !important;
  white-space: nowrap;
}
