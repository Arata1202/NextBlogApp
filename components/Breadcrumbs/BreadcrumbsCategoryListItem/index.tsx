import { Tag } from '@/libs/microcms';
import styles from './index.module.css';

type Props = {
  tag: Tag;
  hasLink?: boolean;
};

export default function PanTagListItem({ tag, hasLink = true }: Props) {
  if (hasLink) {
    return (
      <a href={`/category/${tag.id}`} className={styles.tag}>
        <p className={styles.tag}>{tag.name}</p>
      </a>
    );
  }
  return <span className={styles.tag}>{tag.name}</span>;
}
