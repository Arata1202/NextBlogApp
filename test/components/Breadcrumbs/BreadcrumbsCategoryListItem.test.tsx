import { render } from '@testing-library/react';
import BreadcrumbsCategoryListItem from '@/components/Breadcrumbs/BreadcrumbsCategoryListItem';
import type { Tag } from '@/libs/microcms';

describe('BreadcrumbsCategoryListItem', () => {
  test('スナップショット(BreadcrumbsCategoryListItem)', () => {
    const tag: Tag = { name: 'example', id: '1', createdAt: '', updatedAt: '' };
    const { asFragment } = render(<BreadcrumbsCategoryListItem tag={tag} hasLink={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
