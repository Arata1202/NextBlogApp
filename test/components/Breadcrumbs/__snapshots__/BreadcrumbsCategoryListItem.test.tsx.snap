// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbsCategoryListItem スナップショット(BreadcrumbsCategoryListItem) 1`] = `
<DocumentFragment>
  <a
    class="tag"
    href="/category/1"
  >
    <p
      class="tag"
    >
      example
    </p>
  </a>
</DocumentFragment>
`;
